### PR TITLE
fix: compatible the release name contain slashes

### PR DESCRIPTION
**Pull request by nanjingfm**
## 🚨 Proposed changes

There are two changes in this pr:
1. the release card supports the releases that contain the slashes, e.g. @xx/bb@v0.0.1.
2. sort releases by released data

image

## ⚙️ Types of changes

What types of changes does your code introduce? _Put an `x` in the boxes that apply_

-   [ ] New feature (non-breaking change which adds functionality)
-   [x] Bugfix (non-breaking change which fixes an issue)
-   [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
-   [ ] Documentation Update (if none of the other choices apply)
-   [ ] Refactor

### CODE REVIEW

File: packages/gitlab/src/components/widgets/ReleasesCard/ReleasesCard.tsx [hunk 1]
```diff
@@ -166,15 +163,7 @@ export const ReleasesCard = (props: ReleasesCardProps) => {
             });
     }
     // sort the remaining releases in descending order (latest release first)
-    releases = releases
-        .sort((a, b) => {
```
Comment (nanjingfm) [inline packages/gitlab/src/components/widgets/ReleasesCard/ReleasesCard.tsx:170]: It's duplicated here and in line 148

**Comment by nanjingfm** [comment]
@antoniomuso Looking forward to your feedback, thanks!

**Comment by antoniomuso** [comment]
Thanks, @nanjingfm. Currently, I am on vacation. I will take a look on Monday.

**Comment by nanjingfm** [comment]
> vacation

Okay, have a nice vacation!

**Comment by nanjingfm** [comment]
> Thanks, @nanjingfm. Currently, I am on vacation. I will take a look on Monday.

Okay, have a nice vacation!

File: packages/gitlab/src/components/widgets/ReleasesCard/ReleasesCard.tsx [hunk 2]
```diff
@@ -145,11 +144,9 @@ export const ReleasesCard = (props: ReleasesCardProps) => {
         value?.releases
             .filter(makeFilter(show))
             .sort((a, b) => {
-                try {
-                    return rcompare(a.tag_name, b.tag_name);
-                } catch (error) {
-                    return a.tag_name > b.tag_name ? 1 : -1;
-                }
+                const d1 = new Date(a.released_at || a.created_at);
```
Comment (antoniomuso) [inline packages/gitlab/src/components/widgets/ReleasesCard/ReleasesCard.tsx:147]: If we change sorting, we must update the comment in line 142.

**Review by antoniomuso — CHANGES_REQUESTED** [review]
LGTM! Only one thing.

File: packages/gitlab/src/components/widgets/ReleasesCard/ReleasesCard.tsx (hunk 2 above)
Comment (nanjingfm) [reply packages/gitlab/src/components/widgets/ReleasesCard/ReleasesCard.tsx:147]: I would like to add sort props to avoid breakchange, what do you think?

Comment (antoniomuso) [reply packages/gitlab/src/components/widgets/ReleasesCard/ReleasesCard.tsx:147]: LGTM!

Comment (antoniomuso) [reply packages/gitlab/src/components/widgets/ReleasesCard/ReleasesCard.tsx:147]: But by default, I would like to maintain the new sorting method.

Comment (antoniomuso) [reply packages/gitlab/src/components/widgets/ReleasesCard/ReleasesCard.tsx:147]: We could add a Select component for the sorting method.

Comment (nanjingfm) [reply packages/gitlab/src/components/widgets/ReleasesCard/ReleasesCard.tsx]: image
image

Selecting the sorting method is optional and is not shown by default.

Comment (antoniomuso) [reply packages/gitlab/src/components/widgets/ReleasesCard/ReleasesCard.tsx]: We can show it by default. WDYT?

Comment (nanjingfm) [reply packages/gitlab/src/components/widgets/ReleasesCard/ReleasesCard.tsx]: I just want to keep the same experience as before the modification. And the default sorting should be in line with what most users expect

Comment (antoniomuso) [reply packages/gitlab/src/components/widgets/ReleasesCard/ReleasesCard.tsx]: The feature slightly changes the user experience and improves the UI. We can maintain it active by default.

Comment (nanjingfm) [reply packages/gitlab/src/components/widgets/ReleasesCard/ReleasesCard.tsx]: Okay, agreed.

I've made changes and look forward to your review again!😀

**Review by antoniomuso — CHANGES_REQUESTED** [review]
LGTM, Very good job, thank you. Only one thing, can you rename the commit in a `feat: ...` and specify what you have done?

**Review by antoniomuso — APPROVED** [review]

**Pull request merged**